schema.json

```
@@ -25,6 +25,11 @@
         { "pattern": "\\.(jpg|jpeg)$" }
       ]
     },
+    "video": {
+      "type": "string",
+      "format": "uri",
+      "pattern": "^(https?:\/\/www.(youtube.com\/watch[?]v=([a-zA-Z0-9_-]{0,}))|(https?:\/\/)(www.)?(youtu.be\/([a-zA-Z0-9_-]{0,}))|https?:\/\/(www.)?vimeo.com\/([0-9]{0,}))"
```

```suggestion
      "pattern": "^(?:https?:\\\/\\\/(?:www\\.)?youtu(?:be\\.com\\\/watch\\?v=|\\.be\\\/)(?:[\\w\\-\\_]*)(?:&(amp;)?[\\w\\?=]*)?)$|^(?:https?:\\\/\\\/(?:www\\.)?vimeo\\.com\\\/(?:channels\\\/(?:\\w+\\\/)?|groups\\\/([^\\\/]*)\\\/videos\\\/|)(?:\\d+)(?:|\\\/\\?))$"
```

- More compact form for YouTube
- Allows for various Vimeos formats
- Full length match
- Fully escaped
- No accidental matches with unescaped `.`